### PR TITLE
Add info about helm3 and migrating from helm2

### PR DESCRIPTION
**Pull request by davidnuzik**
- Provides information explaining that K3s supports helm v2 and v3 as of our v1.17.0 release.
- Links to official documentation which has good instructions on how to leverage helm v2 / v3 and migrate.

Closes https://github.com/rancher/k3s/issues/1269

### CODE REVIEW

**Comment by erikwilson** [comment]
The "Auto-deploying manifests" section should probably also mention or have an example for `helmVersion` in the spec.

**Comment by davidnuzik** [comment]
> What do you need to put in the kubeconfig for Helm to work properly? If there's anything specific required for Helm 3, I'm thinking an example might be useful.

There is nothing specific required for helm v2 or helm v3. This was just a statement to ensure the KUBECONFIG environment variable is set correctly or that you specify the config just like with kubectl. I would think users should understand this very clearly if they use kubectl. I can provide an example if you wish but I don't really think it's needed.

**Comment by catherineluse** [comment]
Oh I see. At first I thought kubeconfig was referring to the whole file.

Could this phrase

> Just be sure you have properly set your KUBECONFIG

be changed to

> Just be sure you have properly set your KUBECONFIG environment variable

in the two places where it appears? Otherwise, LGTM.

**Comment by davidnuzik** [comment]
Sure can do this. Thanks!

**Comment by erikwilson** [comment]
It might be good to explicitly show the most usages, for example setting the environment variable:
```
export KUBECONFIG=/etc/rancher/k3s/k3s.yaml
helm ls --all-namespaces
```
or to specify the kubeconfig file per command:
```
helm --kubeconfig /etc/rancher/k3s/k3s.yaml ls --all-namespaces
```

**Comment by davidnuzik** [comment]
Sure will do

File: content/k3s/latest/en/advanced/_index.md [hunk 1]
```diff
@@ -18,6 +19,17 @@ This section contains advanced information describing the different ways you can
...
+
+K3s release _v1.17.0+k3s.1_ added support for Helm 3. You can access the Helm 3 documentation [here](https://helm.sh/docs/intro/quickstart/).
+Note that Helm 3 no longer requires tiller and the helm init command. Refer to the official documentation for details.
+
+K3s does not require any special configuration to start using Helm 3. Just be sure you have properly set your KUBECONFIG for Helm to work properly.
+
+### Upgrading
+
+If you were using Helm v2 in previous versions of K3s, you may upgrade to v1.17.0+k3s.1 or newer and Helm 2 will still function. If you wish to migrate to Helm 3, [this](https://helm.sh/blog/migrate-from-helm-v2-to-helm-v3/) blog post by Helm explains how to use a plugin to successfully migrate. Refer to the official Helm 3 documentation [here](https://helm.sh/docs/) for more information. K3s will handle either Helm v2 or Helm v3 as of v1.17.0+k3s.1. Just be sure you have properly set your KUBECONFIG for Helm to work properly.
+
```
Comment (ShylajaDevadiga) [inline content/k3s/latest/en/advanced/_index.md]: Do we need to mention about setting KUBECONFIG here again?

Comment (davidnuzik) [reply content/k3s/latest/en/advanced/_index.md]: Already to be addressed as per https://github.com/rancher/docs/pull/2165#issuecomment-571225154 will be in incoming commit.

**Comment by davidnuzik** [comment]
Concerns fixed in new commit @erikwilson @catherineluse if you want to give it another quick look.

**Comment by catherineluse** [comment]
It seems like the kubeconfig commands are something that apply for multiple situations beyond just installing Helm 3, so I'm thinking they can have their own section. Then they can be linked to from a variety of places. 

Feel free to wordsmith this:

> This section contains advanced information describing the different ways you can run and manage K3s:
> 
> - [Using Helm 3](#using-helm-3)
> - [Upgrading](#upgrading)
> - [Setting up the kubeconfig file](#setting-up-the-kubeconfig-file)
> 
> …
> 
> # Using Helm 3
> 
> K3s release _v1.17.0+k3s.1_ added support for Helm 3. You can access the Helm 3 documentation [here](https://helm.sh/docs/intro/quickstart/).
> Note that Helm 3 no longer requires tiller and the helm init command. Refer to the official documentation for details.
> 
> K3s does not require any special configuration to start using Helm 3. Just be sure you have properly set up the KUBECONFIG environment variable or specify the location of the kubeconf file with the `--kubeconfig` flag. For details, refer to the section on [Setting up the kubeconfig file.](#setting-up-the-kubeconfig-file)
> 
> # Upgrading
> 
> If you were using Helm v2 in previous versions of K3s, you may upgrade to v1.17.0+k3s.1 or newer and Helm 2 will still function. If you wish to migrate to Helm 3, [this](https://helm.sh/blog/migrate-from-helm-v2-to-helm-v3/) blog post by Helm explains how to use a plugin to successfully migrate. Refer to the official Helm 3 documentation [here](https://helm.sh/docs/) for more information. K3s will handle either Helm v2 or Helm v3 as of v1.17.0+k3s.1. Just be sure you have properly set up your kubeconfig as per the examples in the [section on setting up the kubeconfig file.](#setting-up-the-kubeconfig-file)
> 
> 
> # Setting Up the kubeconfig File
> 
> The kubeconfig file is used to configure access to your Kubernetes cluster. It is required to be set up properly for upgrades, installing applications with Helm charts, 
> 
> Leverage the KUBECONFIG environment variable:
> 
> ```
> export KUBECONFIG=/etc/rancher/k3s/k3s.yaml
> helm ls --all-namespaces
> ```
> 
> Or specify the location of the kubeconfig file per command:
> 
> ```
> helm --kubeconfig /etc/rancher/k3s/k3s.yaml ls --all-namespaces
> ```

**Comment by catherineluse** [comment]
LGTM

**Comment by davidnuzik** [comment]
Will merge in when K3s v1.17.0+k3s.1 is released.

**Pull request merged**